### DISCUSSION TITLE
RSSI after converting to Bridge

### DISCUSSION THREAD

**Discussion by ptr727** [RobertD502/home-assistant-flair#82] [Q&A]
I replaced three gateway pucks with a bridge, and the vent RSSI reported in HA does not appear to update, or at least is not tracking RSSI in the Flair portal.

Any ideas?

**Comment by RobertD502** [comment]
The flair portal as in the history graph of vent RSSI from within the Flair mobile app?

**Reply by ptr727** [reply]
Flair web graph vs. HA graph, here are some screenies:

![image](https://github.com/RobertD502/home-assistant-flair/assets/2061579/afd9b1e1-f816-4a00-8602-0bd964638657)

![image](https://github.com/RobertD502/home-assistant-flair/assets/2061579/808a336c-2ce3-4fc6-bec7-ce8a57449404)

**Reply by RobertD502** [reply]
> Flair web graph vs. HA graph, here are some screenies:
> 
> 
> 
> ![image](https://github.com/RobertD502/home-assistant-flair/assets/2061579/afd9b1e1-f816-4a00-8602-0bd964638657)
> 
> 
> 
> ![image](https://github.com/RobertD502/home-assistant-flair/assets/2061579/808a336c-2ce3-4fc6-bec7-ce8a57449404)
> 
> 

Have you attempted to restart Home Assistant or reload the Flair integration?

**Reply by ptr727** [reply]
I have, but let me do it again for good measure.

**Reply by ptr727** [reply]
![image](https://github.com/RobertD502/home-assistant-flair/assets/2061579/e23b9834-f272-4576-9efc-4e0518ed0fb1)

![image](https://github.com/RobertD502/home-assistant-flair/assets/2061579/1a7d036a-edf2-4296-92b6-3f1f441549fa)

This restart changed something in the observed behavior.
First HA restart was after bridge add and remove pucks, I then moved bridge to new location, but did not restart HA, maybe there is some opportunity to investigate robustness of reporting?

**Reply by RobertD502** [reply]
> ![image](https://github.com/RobertD502/home-assistant-flair/assets/2061579/e23b9834-f272-4576-9efc-4e0518ed0fb1)
> 
> 
> 
> ![image](https://github.com/RobertD502/home-assistant-flair/assets/2061579/1a7d036a-edf2-4296-92b6-3f1f441549fa)
> 
> 
> 
> This restart changed something in the observed behavior.
> 
> First HA restart was after bridge add and remove pucks, I then moved bridge to new location, but did not restart HA, maybe there is some opportunity to investigate robustness of reporting?

So, the integration has no control when it comes to anything that is reported. Take for example, the RSSI value - for each vent, the integration polls the current-state API endpoint on Flair's servers (where the RSSI value is one of the things reported). In Home Assistant, the RSSI sensor will have a state equal to whatever the Flair API returned. In your case, the Flair API was returning the same value over and over again during the time period that you see the RSSI value as being continuously the same. We don't have a way of making the reporting more robust since Flair's servers are fully in charge of the reporting. The Flair API gets polled every 30 seconds and the integration in Home Assistant reports back what was returned in the form of entities.

What likely happened is, when you moved the bridge and it reestablished its connection to Flair's servers, the current-state endpoint started to function properly again.

**Reply by ptr727** [reply]
Understood, still strange that the ~3:30pm period when reporting started working again was when I restarted HA again, while I moved the bridge ~1pm, and the HA observed values did not change when I moved the bridge. So maybe the reconnecting to Flair API's had more to do with it than the bridge power cycle event?

**Reply by RobertD502** [reply]
> Understood, still strange that the ~3:30pm period when reporting started working again was when I restarted HA again, while I moved the bridge ~1pm, and the HA observed values did not change when I moved the bridge. So maybe the reconnecting to Flair API's had more to do with it than the bridge power cycle event?

This particular situation could have been a caching issue, server backend issue, etc. Since we, as end-users, don't have any insights into the Flair servers, it won't be possible to definitively know what caused the issue you faced.
